### ISSUE TITLE
Newsletter Blog RSS

### ISSUE THREAD

**Issue by kuunda** [mautic/mautic#1116]
hey all.

I was wondering is there a way to replicate RSS new blog post feed that you get in mailchimp in Mautic without using MailChimp?

The idea is to have RSS feed/reader setup from Wordpress listing the blog content, images, links, authors etc that can be mapped and pulled into the new email as variable fields.

The aim is to send a new newsletter blog post each time you post a new article.

Step 1. Typically when someone signs up to your blog widget you send them a double optin email (Mautic form this isn't a problem, just not sure how to double optin them in i.e. Send an email with a link once the user has validated the link confirm optin)

Step 2. Then read RSS feed of the most recent blog post looking for the newest blog post - this the part that I am hoping someone has considered on the best way to deliver this.
1. Thereafter the RSS feed into Mauctic can manage the new blog post is posted and automatically sends to everyone still opted in.

I know I asking for Ethiopia but if someone has started something like this please let me know, maybe we can work together on getting this to work

**Comment by npracht** [comment]
Hi,

This is an excellent feature request. Often asked by people and you could save a lot of time.
Also call **RSS to Email**.

:+1: 

N.

**Comment by dmarkotter** [comment]
The MailPoet plugin for Wordpress allows you to curate a newsletter from (multiple) blog posts - it'll also do it automatically, say once a week.
https://support.mailpoet.com/knowledgebase/send-your-latest-blog-posts-to-your-subscribers/

**Comment by healthstatus** [comment]
I would love to see this feature.

**Comment by phillipwilhelm** [comment]
+1 for this idea

**Comment by OlafDammann** [comment]
Yupp, I'd love to see it, too :) Especially as I have some people searching for a new Tool (which doesn't host data in the US) which use RSS-Mail exessively.
Happy new year to everyone here!
OLAF

**Comment by 3ruce** [comment]
+1 - woud be great!

**Comment by npracht** [comment]
Hi @dbhurley ,

We're probably gonna work on it. I have few questions before developing:
- Did you start working on that feature? I don't either you or us waste time working on the same feature.
- Do you have any advice if we should do it directly as core feature or via a plugin?

In that case there isn’t third party application (it refers only to RSS feeds which are market standards).
—> I was thinking it should be core.

Thanks for your advices,

N.

**Comment by healthstatus** [comment]
We have this in alpha testing now.   Would you like a copy to play with?

 

 

Greg White

HealthStatus LLC

 

888.726.4683 x604

317.823.2687 x604

 

HealthStatus Consumer Portal:  <http://www.healthstatus.com/> HealthStatus.com

HealthStatus Assessment Software:  <http://hra.net/> HRA.net

**Comment by npracht** [comment]
@healthstatus ,

That would be a great idea. Please share and instead of developing we share our feedbacks and we test your development?

N.

**Comment by npracht** [comment]
@healthstatus , how would you proceed to send me your development? My devs have time next week for testing :-)

**Comment by healthstatus** [comment]
My guys are putting together a zip file (with a TINY amount of documentation), hopefully ready by end of day tomorrow, worst case.

**Comment by npracht** [comment]
Great!

**Comment by healthstatus** [comment]
[FeedManBundle.zip](https://github.com/mautic/mautic/files/251115/FeedManBundle.zip)
Here is our plugin bundle.

install manually. You need to copy the FeedManBundle in plugins folder, rename prod folder in  app/cache folder (or delete cache) and then click the Install/Upgrade plugins button in the Plugin manager in admin panel.

Like I said, ALPHA, no support plans at this point, but we are retaining all rights for now.  Comments are welcome, code updates are welcome.

**Comment by npracht** [comment]
Hi @healthstatus 

I'll ask my dev team to install that and we'll have a look on it.
Afterward, i'll send you my feedbacks concerning the use and features.

Talk to you soon,

N.

**Comment by npracht** [comment]
Hi @healthstatus,

I have few questions to use it (i didn't find any documentation) and how to contribute.
Could you send me your skype ID or phone number by email (you have it here: https://github.com/npracht) to have a talk with you please?
I wait for your email :-)

N.

**Comment by fmenossi83** [comment]
Hi, i have installed but nothing happened. I did everything you told me. is there anything else?

**Comment by ptcc** [comment]
👍

**Comment by hoodaajay99** [comment]
This will be an awesome plugin to have with Mautic, Any updates when this will be production ready?

**Comment by npracht** [comment]
Hi @hoodaajay99 we're now upgrading it to 2.0.1 (we have it under an older version. It should come in a couple of weeks.

N.

**Comment by ninjoan** [comment]
Any update about this topic.

**Comment by npracht** [comment]
Hello @ninjoan,

Due to several changes with the 2.0 and 2.1, we're late on that topic (we've adapted and pull other topics first).
We have something usable in 1.4.x, we need to bring it to the last version. I guess that'll be done in September/October latest.

i'll let you know.

N.

**Comment by valmirbispo** [comment]
I try to install the "FeedManBundle.zip", but received this message
![sem titulo](https://cloud.githubusercontent.com/assets/1265459/18051543/957d9e3c-6dcb-11e6-9c3d-5db37110c067.png)

**Comment by escopecz** [comment]
@valmirbispo thanks for the screenshot, but it doesn't say much about the actual problem. Please, read https://mautic.org/docs/en/tips/troubleshooting.html for more about the error reporting and let us know what the error message in the logs is.

**Comment by MaxWebmecanik** [comment]
Hi John, we've been working on infrastructure and rethinking our strategy these last 2 months so our prototype for RSS to Email ended up in arrested development.
In the coming weeks, I'm going to do my best to bring the existing codebase up to date with the latest Mautic development version and share it in its incomplete form as a pull request to get some feedback and continue working on it to have a newer working prototype ASAP.

**Comment by valmirbispo** [comment]
@escopecz thanks for the help. I saw the following error log:
[2016-09-01 11:28:23] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\TableNotFoundException: "An exception occurred while executing 'SELECT f0_.id AS id_0, f0_.name AS name_1, f0_.url AS url_2, f0_.leadlist_id AS leadlist_id_3, f0_.schedule_day AS schedule_day_4, f0_.send_feed AS send_feed_5, f0_.send_feed_type AS send_feed_type_6, f0_.status AS status_7, f0_.date AS date_8 FROM feed f0_ ORDER BY f0_.id DESC':  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist" at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 53 {"exception":"[object] (Doctrine\\DBAL\\Exception\\TableNotFoundException(code: 0): An exception occurred while executing 'SELECT f0_.id AS id_0, f0_.name AS name_1, f0_.url AS url_2, f0_.leadlist_id AS leadlist_id_3, f0_.schedule_day AS schedule_day_4, f0_.send_feed AS send_feed_5, f0_.send_feed_type AS send_feed_type_6, f0_.status AS status_7, f0_.date AS date_8 FROM feed f0_ ORDER BY f0_.id DESC':\n\nSQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:53, Doctrine\\DBAL\\Driver\\PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:106, PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:104)"} []
[2016-09-01 11:29:21] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\TableNotFoundException: "An exception occurred while executing 'SELECT f0_.id AS id_0, f0_.name AS name_1, f0_.url AS url_2, f0_.leadlist_id AS leadlist_id_3, f0_.schedule_day AS schedule_day_4, f0_.send_feed AS send_feed_5, f0_.send_feed_type AS send_feed_type_6, f0_.status AS status_7, f0_.date AS date_8 FROM feed f0_ ORDER BY f0_.id DESC':  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist" at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 53 {"exception":"[object] (Doctrine\\DBAL\\Exception\\TableNotFoundException(code: 0): An exception occurred while executing 'SELECT f0_.id AS id_0, f0_.name AS name_1, f0_.url AS url_2, f0_.leadlist_id AS leadlist_id_3, f0_.schedule_day AS schedule_day_4, f0_.send_feed AS send_feed_5, f0_.send_feed_type AS send_feed_type_6, f0_.status AS status_7, f0_.date AS date_8 FROM feed f0_ ORDER BY f0_.id DESC':\n\nSQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:53, Doctrine\\DBAL\\Driver\\PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:106, PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vfbne059_mautic.feed' doesn't exist at /home/vfbne059/public_html/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:104)"} []

**Comment by badri** [comment]
@valmirbispo After installing the plugin, you will have to run the following commands:

one for clearing the app cache,

`$ php app/console cache:clear --env=prod`

another for creating tables,

`$ php app/console doctrine:schema:update --env=prod --force`

Please confirm if you still get these errors after running the above commands.

@healthstatus I've installed the plugin successfully. Any docs on how it works? Thanks.

**Comment by BatchHeader** [comment]
Running fresh install of Mautic 2.2.1 on PHP7.0 
I have

- installed plugin
- cleared cache
- updated plugins
- Setup single feed (Daily,Last Five, Active)

Now when I try to enter any compose email screen I receive a 500 server error with the following in my logs. Happy to help troubleshoot but I don't know enough of Mautic insides to know where to start.

`[2016-10-24 18:22:57] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Type error: Argument 1 passed to MauticPlugin\FeedManBundle\EventListener\EmailSubscriber::__construct() must be an instance of Mautic\CoreBundle\Factory\MauticFactory, none given, called in /home/[REDCATED]/public_html/tellmemore/app/cache/prod/appProdProjectContainer.php on line 5378" at /home/[REDCATED]/public_html/tellmemore/plugins/FeedManBundle/EventListener/EmailSubscriber.php line 21 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Type error: Argument 1 passed to MauticPlugin\\FeedManBundle\\EventListener\\EmailSubscriber::__construct() must be an instance of Mautic\\CoreBundle\\Factory\\MauticFactory, none given, called in /home/[REDCATED]/public_html/tellmemore/app/cache/prod/appProdProjectContainer.php on line 5378 at /home/[REDCATED]/public_html/tellmemore/plugins/FeedManBundle/EventListener/EmailSubscriber.php:21)"} []`

**Comment by badri** [comment]
@BatchHeader You will have to remove the constructor where the error occurs. Plus you will hit upon a lot of other snags. I'm entirely rewriting this plugin and will be sending a PR shortly.

Some points of note on the implementation:
1. The Feedman tab will be as before, where users will be able to add feeds, set an interval and a segment to which the feed will be sent.
2. There will be 2 cron jobs, 
a. 1 which will pick up feed data and update it in the DB.
b.  another which will send the updated feeds to the configured segments.

For the initial version, 
1. There won't be any campaign builder widget related to feeds.
2. No Feed related tokens, ex. http://www.activecampaign.com/help/how-to-use-rss-in-your-email-messages-campaigns/, though I'd love to add them in future.

I'm not sure if we will need an EmailSubscriber for this plugin, at least not until #2 above is implemented.

**Comment by BatchHeader** [comment]
@badri That all sounds great; I'll keep an eye out. Give me a shout if you need any testing either before or after PR.

**Comment by hilltopmediadesign** [comment]
@badri thanks for working on this. Much needed. Count me in for testing.

**Comment by npracht** [comment]
Hi guys,

I've linked a PR with the feature almost finished. It just misses a bit of help to finish it cause I don't have the dev resources to finish it unfortunately.
Feel free to take that in charge.

N.

**Comment by ninjoan** [comment]
What dev you need @npracht

**Comment by npracht** [comment]
Hi @ninjoan check PR #2888 i've done. We're almost done but we internally have some issues my dev cannot work on it anymore. I let you have a look on that and if you need more information, i'll ask my dev to have a talk with you.

It'd be awesome that you can finish this dev :)

**Comment by SvetZitrka** [comment]
Are there any news regarding this extension?

**Comment by ladislavsulc** [comment]
+ 1, this one is really important, any news? Thank you.

**Comment by npracht** [comment]
As i said i cannot carry it anymore. If some dev wanna finish it it's not so much to do anymore.

**Comment by taropaa** [comment]
For what it's worth I am throwing in my +1, as this would definitely be a useful feature.

**Comment by afanjul** [comment]
Very useful feature, any updates on this?

**Comment by Sattvic** [comment]
I think this should be part of core.  Maybe it can start as a plugin but eventually should become part of Mautic's core updates and feature set.  For me, this is a critical aspect of Digital Marketing Automation using RSS technology which is older but still appropriate for custom marketing solutions.  This is how Mailchimp approaches it:
https://mailchimp.com/features/rss-to-email/

Mailchimp does not care how the RSS Feed is created (wordpress creates a feed by default).  In my particular case, I custom created a feed against my database to be in a standard RSS format.  Mautic can be configured in a similar way...just enter a valid RSS Feed URL (https://validator.w3.org/feed/), set the email frequency, combine into a pre-existing template/layout and run infinitely.

Here are a couple of use cases of using RSS for Automation Marketing:
- eCommerce stores: custom RSS feeds can be configured to update specific audience segments once they are adding to the online store for purchase
- eTickets: Real-Time email updates to segmented audience that are opted-in to receive updates on a filtered genre
- Real Estate: Auto email alerts once a new real estate listing is added
- Event Websites: TimeOutNewYork (for example) - signup for daily/weekly emails on things to do in your city
Pet Adoption: Help automatically spread the word once an animal becomes available for fostering or adoption, etc.
Groupon: Receive auto-emails whenever a new deal is added to the database based on your preferences

**Comment by npracht** [comment]
Hi @Sattvic if you can take the lead on that PR https://github.com/mautic/mautic/pull/2888 that has not been finished. most of the work was done. Unfortunately my team cannot handle it anymore because of a lack of time.

**Comment by afanjul** [comment]
Finally we installed and used Mailtrain due to the lack of RSS support in Mautic, maybe this can help to somebody...

**Comment by gingerling** [comment]
+1

**Comment by dataaegissoftware** [comment]
+1

**Comment by moskoweb** [comment]
Hello everyone, I have developed a plugin for WordPress, which integrates with Mautic, allowing each new post to do so, you can configure to send an email through Mautic to a certain thread for you.

Currently it is only in Portuguese Brazil, but I am already translating it, I would like to know what you feel would be of more importance to have in a plugin of this?

Who wants to know more about what I already have, you can see here: [https://lp.alanmosko.com.br/plugin-mautic-rss/](https://lp.alanmosko.com.br/plugin-mautic-rss/)

**Comment by dbhurley** [comment]
Seems there are several plugins now created to handle this functionality. Closing the issue.

**Comment by gingerling** [comment]
Can you link to all the plugins here please, for reference. I can only find the Portuguese Brazil one linked above.

**Comment by 3ruce** [comment]
Likewise, I have no idea where the plugins can be found?

https://www.mautic.org/integrations/ implies that rss is built in, but I can't find any documentation or anything in the app itself that suggests rss and mautic are on the same wavelength...

What am I missing?

**Comment by ChrisRAoW** [comment]
I created a plugin which can create an email from a rss-feed. Would be nice if some people can test it:

https://github.com/ChrisRAoW/mautic-rss-to-email-bundle

**Comment by 3ruce** [comment]
I will be testing this out for sure, I just need a week or two! Thank you, it looks great!

**Comment by Dcoutelle** [comment]
Installation way look nice if this is not merged into mautic core 😃

**Comment by gingerling** [comment]
Hi, I would like to test this, can you point me at install instructions, thanks

**Comment by ChrisRAoW** [comment]
@gingerling You need to have the self hosted mautic install. Check if you have a composer.json in the root of the mautic installation. If you haven't make sure you download it first from the github of mautic. You have to download composer first if you do not have it. Rest of the installation instructions are in the READme of the Github repo: https://github.com/ChrisRAoW/mautic-rss-to-email-bundle

**Comment by justlevine** [comment]
While @ChrisRAoW was nice enough to create a plugin that lets you bring in RSS content into your email (much kudos!!!!), we're still lacking the ability to _trigger_ emails based on RSS activity.
As per the [blog](https://www.mautic.org/community-blog/moving-feature-requests/), Ive reopened this feature request in the [Ideas Forum](https://www.mautic.org/topic/rss-based-email-automation-feature-request/).

**Comment by dbhurley** [comment]
You're the best @justlevine! Thank you.

**Comment by devsrealm** [comment]
Unfortunately, the RSS plugin by ChrisRAoW doesn't work with mautic 3.0 and above

**Comment by juicerocket** [comment]
Any update on this functionality? Does M3 allow RSS--> email?

**Comment by ChrisRAoW** [comment]
@juicerocket I will try to make my plugin compatible with M3 soon.

**Comment by justlevine** [comment]
@ChrisRAoW I and many others appreciate the effort you put into maintaining your plugin. Kudos!

That said, it's rediculous that Mautic hasn't integrated such an essential feature into core.

**Comment by ChrisRAoW** [comment]
@justlevine Great to hear it's appreciated. Probably they didnt had the need to do it because there was such a wonderful plugin ;-)

**Comment by npracht** [comment]
@ChrisRAoW we already made it compatible to M3. You have pending PR [on your repository](https://github.com/ChrisRAoW/mautic-rss-to-email-bundle/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc) :-)

**Comment by ChrisRAoW** [comment]
@npracht Fantastic, I totally forgot this. Thanks, will check it out.

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Community Forums**. There might be relevant details there:

https://forum.mautic.org/t/rss-based-email-automation-feature-request/10274/1

**Comment by woutersf** [comment]
Try this one: 
https://github.com/woutersf/MauticEmailRssImportBundle
